src/hotspot/share/gc/g1/g1ConcurrentMark.hpp

```diff
@@ -136,10 +136,92 @@ class G1CMMarkStack {
     G1TaskQueueEntry data[EntriesPerChunk];
   };
 
-  size_t _max_chunk_capacity;    // Maximum number of TaskQueueEntryChunk elements on the stack.
+  class ChunkAllocator {
+    // The chunk allocator relies on a growable array data structure that allows resizing without the
+    // need to copy existing items. The fundamental approach involves organizing the array into chunks,
+    // essentially creating an "array of arrays"; referred to as buckets in this implementation. To
+    // facilitate efficient indexing, the size of the first chunk is set to a power of 2. This choice
```

Comment:
I believe the "Chunk" On L136 mean sth diff as the one here -- here, you are probably referring to a "bucket".